[TSK-93] Feat: 회원가입 페이지 생성 및 API 연동

gazi52 opened the pull request:
회원가입 페이지 생성 및 API 연동 완료했습니다. 로그인에 필요한 ID/PW는 만들어서 Slack에 올려놓겠습니다.

- 아이디, 닉네임, 비밀번호 규칙 존재
- 회원가입 요청 시 규칙 미적용에 따른 에러파싱 적용 중
- 프로필 사진 적용 필요

Resolves: TSK-93

gazi52 committed:
[TSK-93] Feat: 회원가입 API 연동

회원가입 페이지 제작 및 API 연동
 - 프로필 사진 관련 갤러리 접근 필요
 - Call adapter로 에러파싱 적용 필요

gazi52 committed:
Merge branch 'login-page' of https://github.com/wafflestudio21-5/team2-android into login-page

gazi52 committed:
Merge branch 'main' into login-page

snupinel commented on app/src/main/java/com/wafflestudio/bunnybunny/pages/StartPage.kt:
```diff
@@ -104,8 +103,7 @@ fun StartPage(
-                                val loginRequest = viewModel.tryLogin(idInput, pwInput)
-                                //val ifLogin = viewModel.api.testLogin("Bearer " + loginRequest.token)
+                                val loginRequest = viewModel.tryLogin(emailInput, pwInput)
                                 withContext(Dispatchers.Main){
                                     // TODO onNavigateToMain
                                 }
```
로그인 성공하면 여기서 홈화면으로 넘어가는 거겠네요!

glenn-syj reviewed (approved):
콜 어댑터가 완성된 경우도 궁금해지네요! 작성하시느라 고생 많으셨습니다.

JuTaK97 commented on app/src/main/java/com/wafflestudio/bunnybunny/lib/network/call_adapter/ErrorParsedHttpException.kt:
```diff
@@ -0,0 +1,4 @@
+package com.wafflestudio.bunnybunny.lib.network.call_adapter
+
+class ErrorParsedHttpException {
```
수상할 정도로 익숙한...

JuTaK97 commented on app/src/main/java/com/wafflestudio/bunnybunny/pages/SignupPage.kt:
```diff
@@ -0,0 +1,137 @@
+    var nickname by rememberSaveable { mutableStateOf("") }
+    //var pwInput by rememberSaveable { mutableStateOf("") }
+    Column {
+        BasicTextField(
```
공통된 디자인의 BasicTextField를 여러 번 쓰고 있는데, BasicTextField를 한 단계 래핑해서 모든 요소(색깔, 글꼴, 클릭 이벤트 등등 모든 것)를 파라미터로 받고 공통된 부분을 default parameter로 놓으면 코드가 깔끔해집니다!

JuTaK97 replied:
재사용성을 챙길 수 있어요

glenn-syj committed:
[TSK-101] Design: 앱 로고 추가

앱 로고를 추가했습니다.

- round 버전은 추후 수정 필요합니다.
- 로고에 관해서는 추후 논의가 필요합니다.

glenn-syj committed:
[TSK-104] Feat: 카카오 로그인 기능 구현 및 일반 로그인 로직 추가

- 카카오 로그인 기능 추가
- 404 받은 후 회원가입 분기

### 개선 필요
- 회원 가입 후 오류
- compose 간 idToken 변수 전달과 보안
